aggregator_api/src/lib.rs

```diff
@@ -351,86 +384,147 @@ mod models {
-        pub(crate) task_id: Option<TaskId>,
-        pub(crate) leader_endpoint: Url,
-        pub(crate) helper_endpoint: Url,
+        /// URL relative to which this task's peer aggregator's DAP API can be found. The peer
+        /// aggregator plays the DAP role opposite to the one in the `role` field.
+        pub(crate) peer_aggregator_endpoint: Url,
+        /// DAP query type for this task.
         pub(crate) query_type: QueryType,
+        /// The VDAF being run by this task.
         pub(crate) vdaf: VdafInstance,
+        /// The role that this aggregator will play in this task.
         pub(crate) role: Role,
-        pub(crate) vdaf_verify_key: Option<String>,
+        /// The VDAF verification key used for this DAP task, as Base64 encoded bytes. Task ID is
+        /// derived from the verify key.
+        pub(crate) vdaf_verify_key: String,
+        /// The maximum number of times a given batch may be collected.
         pub(crate) max_batch_query_count: u64,
-        pub(crate) task_expiration: Option<u64>, // seconds since UNIX epoch
+        /// The time after which the task is considered invalid.
+        pub(crate) task_expiration: Option<Time>,
+        /// The minimum number of reports in a batch to allow it to be collected.
         pub(crate) min_batch_size: u64,
-        pub(crate) time_precision: u64, // seconds
+        /// The duration to which clients should round their reported timestamps, as seconds since
+        /// the UNIX epoch.
+        pub(crate) time_precision: Duration,
+        /// HPKE configuration for the collector.
         pub(crate) collector_hpke_config: HpkeConfig,
+        /// If this aggregator is the leader, this is the bearer token to use to authenticate
+        /// requests to the helper, as Base64 encoded bytes. If this aggregator is the helper, the
+        /// value is `None`.
         pub(crate) aggregator_auth_token: Option<String>,
-        pub(crate) collector_auth_token: Option<String>,
     }
 
     #[derive(Clone, Debug, PartialEq, Eq, Serialize, Deserialize)]
     pub(crate) struct TaskResp {
+        /// ID of the DAP Task.
         pub(crate) task_id: TaskId,
-        pub(crate) leader_endpoint: Url,
-        pub(crate) helper_endpoint: Url,
+        /// URL relative to which this task's peer aggregator's DAP API can be found. The peer
+        /// aggregator plays the DAP role opposite to the one in the `role` field.
+        pub(crate) peer_aggregator_endpoint: Url,
+        /// DAP query type for this task.
         pub(crate) query_type: QueryType,
+        /// The VDAF being run by this task.
         pub(crate) vdaf: VdafInstance,
+        /// The role that this aggregator will play in this task.
         pub(crate) role: Role,
-        pub(crate) vdaf_verify_keys: Vec<String>,
+        /// The VDAF verification key used for this DAP task, as Base64 encoded bytes. Task ID is
+        /// derived from the verify key.
+        pub(crate) vdaf_verify_key: String,
```

Comment:
Ideally this structure would not contain a secret value like VDAF verify key, to reduce the risk of it being exposed. However I'm not sure if that breaks task sync.

Reply:
Refresh my memory on what task sync accomplishes? I thought it was to sync up task definitions from janus to control plane. If that's the case then presumably the leader and helper already have the vdaf_verify_key and handing this over is not necessary?

Reply:
I think I agree. But I'm not certain whether `divviup-api` ever needs to see this value beyond provisioning it into the aggregators, so I'm just making a note that we have to think about this before we risk breaking task sync.